docs/cli/app-types.md

```diff
@@ -0,0 +1,141 @@
+# .NET Core CLI types of portable applications
+
+## Introduction
+There are several ways to think about the "types" of applications you can build. Usually, the types describe 
+a certain execution model or are based on what the application can "do"; examples of these are "console application", 
+"web application" etc. All of these types of applications (and more) can be created with .NET Core, since it is 
+a general purpose development platform.
+
+However, given its unique and cross-platform nature, .NET Core also has another angle through which to observe the type of the application 
+and that is application's *portability*. Portability here means, essentially, where can you run your application and what are the 
+pre-requisites that you need to satisfy in order for your application to run on a given machine. This document 
+deals with this axis, portability, and outlines the two main types of portability that .NET Core enables. 
+
+There are two main types that we can observe: 
+
+1. Portable application
+    * As a sub-category of this we have the portable application with native dependencies
+2. Self-contained application
+
+## Portable application
+Fully portable application is an intermediate language (IL) application that targets a shared component called **the shared runtime**. 
+This makes it portable to any platform (operating system) that the shared runtime runs on. 
+
+This type of application will only carry its own code and dependencies that are outside of the shared runtime. 
+As an example, let's say you are making a console application that has the ability to invoke a certain REST API 
+and deserialize the returned JSON into a type and then display it. You have everything you need for this small 
+application expect for a good JSON parser; for this, you add a dependency to your `project.json` to include 
+[Json.NET](https://www.nuget.org/packages/Newtonsoft.Json/). Once you publish your application using `dotnet publish`, 
+you will see that only your application's code and JSON.net have been published in the output. 
+The shared runtime remains outside of your application's dependency closure. 
+
+Of course, this type of application requires you to have a shared runtime installed. Otherwise, you will not be able 
+to run the application. 
+
+In order to create a portable application, the only thing you need is to target the shared runtime in your `project.json`
```

Comment:
since you have to actually do two things, I wouldn't say the only thing and would remove of course from here too. Maybe rewrite as "In order to create a portable application, all you have to do is to target the shared runtime in your `project.json` and have your frameworks aligned as shown in the following example."